pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,431 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+Implementation of the BaseRepository abstract classes that defines an in-memory
+repository based on dictionaries.
+"""
+
+from __future__ import absolute_import, print_function
+
+import six
+
+from pywbem import CIMClass, CIMQualifierDeclaration, CIMInstance
+
+from pywbem._nocasedict import NocaseDict
+from pywbem._utils import _format
+
+from ._baserepository import ObjectStoreAPI, BaseRepositoryAPI
+from ._utils import _uprint
+
+__all__ = ['InMemoryRepository']
+
+
+class InMemoryObjStore(ObjectStoreAPI):
+    """
+    Derived from the :class:`ObjectStoreAPI`, this class implements a
+    dictionary based in-memory repository for CIM objects that manages
+    CIM classes, CIM instances, and CIM qualifier declarations.
+    """
+
+    def __init__(self, case_insensitive_names, cim_object_type):
+        """
+          Parameters:
+
+            case_insensitive_names(:class:`py:bool`):
+                Defines whether the data store for this type handles
+                case insesitive names or not
+
+            cim_object_type ():
+             CIMType of the object stored in this object store. Only CIMClass,
+             CIMInstance, and CIMQualifierDeclaration types are allowed.
+        """
+
+        super(InMemoryObjStore, self).__init__(case_insensitive_names,
+                                               cim_object_type)
+
+        self._data = NocaseDict() if case_insensitive_names else {}
+
+    def __repr__(self):
+        return _format('InMemoryObjStore(type={0},  dict={1}, size={2}',
+                       self._cim_object_type, type(self._data),
+                       len(self._data))
+
+    def exists(self, name):
+        """
+        Overrides the abstract method.
```

Comment:
Is this really needed? For a user of the class, it does not matter.
Occurs multiple times.

Reply:
Agreed. Removed.  I would really like to   just reference the baserepository documentation from the inmemory repository rather than repeat it.  I changed the documentation for all of the methods in the inmemory file to say, for example:

See :meth:`~pywbem_mock.BaseObjectStore.len`

or
See :meth:`~pywbem_mock.BaseRepository.validate_namespace`

so that we do not repeat the documentation:

DISCUSSION: Is this valid for you

Reply:
Having a one-line sentence that links to the base class description with a clickable link is fine with me.

However, can you give the custom-inherit package a try? If that works out fine, I would prefer that.

Reply:
Done.  All method doc is in the base class.  Note that there are no strings at all for the methods of the derived class, not even """""" That works and does not even cause pylint, etc. to raise a message.  I had thought of just putting and empty doc string for each but do not see the need. I did put a comment at the class level about the lack of documentation.support in ne

It was a PIA to get the right form that works for both python 2 and 3 and pinned the requirements.txt to a specific version because the developer is planning to remove python 2.7
support in next version.

Reply:
Ok.